src/execution/operator/csv_scanner/sniffer/header_detection.cpp

```diff
@@ -93,9 +93,51 @@ static string NormalizeColumnName(const string &col_name) {
 	}
 	return col_name_cleaned;
 }
+
+// If our columns were set by the user, we verify if their names match with the first row
+bool CSVSniffer::DetectHeaderWithSetColumn() {
```

Comment:
Could we add a test that verifies this works correctly? The way I understand this is that `COPY` should always round-trip now, e.g.:


```sql
CREATE TABLE my_varchars(a VARCHAR, b VARCHAR, c VARCHAR);
INSERT INTO my_varchars VALUES ('Hello', 'Beautiful', 'World');
COPY my_varchars TO 'varchar_header.csv' (HEADER 1);
COPY my_varchars TO 'varchar_no_header.csv' (HEADER 0);

COPY my_varchars FROM 'varchar_header.csv';
COPY my_varchars FROM 'varchar_no_header.csv';
FROM my_varchars;
┌─────────┬───────────┬─────────┐
│    a    │     b     │    c    │
│ varchar │  varchar  │ varchar │
├─────────┼───────────┼─────────┤
│ Hello   │ Beautiful │ World   │
│ Hello   │ Beautiful │ World   │
│ Hello   │ Beautiful │ World   │
└─────────┴───────────┴─────────┘
```